src/swim/swim.c

```diff
@@ -116,8 +118,36 @@ swim_updates_send(struct swim_context *ctx, swim_id_t id, swim_id_t to)
 out_unlock:
 	swim_ctx_unlock(ctx);
 
-	if (rc == 0)
-		rc = ctx->sc_ops->send_message(ctx, to, upds, i);
+	if (rc == 0) {
+		nupds = i;
+		if (d_log_check(DLOG_INFO | D_LOGFAC)) {
+			FILE *fp;
+			char *msg;
+			size_t msg_size;
+
+			fp = open_memstream(&msg, &msg_size);
+			if (fp != NULL) {
+				for (i = 0; i < nupds; i++) {
+					rc = fprintf(fp, " {%lu %c %lu}",
+					     upds[i].smu_id,
+					    "ASD"[upds[i].smu_state.sms_status],
+					     upds[i].smu_state.sms_incarnation);
+					if (rc < 0)
+						break;
+				}
+
+				fclose(fp);
+				/* msg and msg_size will be set
+				 * after fclose(fp) only
+				 */
+				if (msg_size > 0)
+					SWIM_INFO("%lu ==> %lu:%s\n",
+						  self_id, to, msg);
+				free(msg); /* allocated by open_memstream() */
+			}
```

Comment:
what if fp==null?

Reply:
this is inner in if (fp != NULL). So, fp should NOT be NULL here.